pkg/transform/kubernetes/populators_test.go

```diff
@@ -588,6 +588,53 @@ func TestPopulateEnvFrom(t *testing.T) {
 	}
 }
 
+func TestPopulateVolumes(t *testing.T) {
+	volumeClaims := []spec.VolumeClaim{{Name: "foo"}, {Name: "bar"}, {Name: "barfoo"}}
+	volumes := []api_v1.Volume{{Name: "foo"}}
+
+	// a volumeMount is defined but that is not there in volumeClaims
+	// neither it is in pod level volumes, so this should fail
+	failingContainers := []api_v1.Container{
+		{VolumeMounts: []api_v1.VolumeMount{{Name: "baz"}}},
+	}
+
+	if _, err := populateVolumes(failingContainers, volumeClaims, volumes); err == nil {
+		t.Errorf("should have failed but passed for volumeMount that" +
+			" does not exist.")
+	}
+
+	passingContainers := []api_v1.Container{
```

Comment:
Also from Kubernetes -
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/run_test.go#L32
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/run_test.go#L406
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/run_test.go#L648
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/run_test.go#L648
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/run_test.go#L741
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/run_test.go#L834
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/run_test.go#L925
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/run_test.go#L1023
and tons more.